components/script/dom/permissions.rs

```diff
@@ -268,14 +267,10 @@ impl PermissionAlgorithm for Permissions {
             PermissionState::Prompt => {
                 let perm_name = status.get_query();
 
-                let globalscope = GlobalScope::current().expect("No current global object");
-
                 // https://w3c.github.io/permissions/#request-permission-to-use (Step 3 - 4)
-                let state = prompt_user(
-                    &format!("{} {} ?", REQUEST_DIALOG_MESSAGE, perm_name.clone()),
-                    globalscope.is_headless(),
-                );
-
+                let globalscope = GlobalScope::current().expect("No current global object");
+                let prompt = format!("{} {} ?", REQUEST_DIALOG_MESSAGE, perm_name.clone());
```

Comment:
Rather than doing string formatting for dialogs in this code, I think we should delegate that to the embedder and only send an enum value that represents the kind of prompt required.